### ISSUE TITLE
dbservers stuck on default storageClassName standard instead of the deployment specification

### ISSUE THREAD

**Issue by mizunos** [arangodb/kube-arangodb#525]
Operator v 0.4.4
Arangodb v 3.6.1 running on GKE (enterprise edition)
I have the following definition for an arango single server in ActiveFailover mode and a storageClass object definition for ssd

```
  agents:
    nodeSelector:
      dbtype: arango
    count: 3
    args:
      - --log.level=debug
    storageClassName: ssd
    resources:
      requests:
        storage: 8Gi
  dbservers:
    nodeSelector:
      dbtype: arango
    count:  2
    storageClassName: ssd
    resources:
      requests:
        storage: 30Gi
  externalAccess:
    type: None
```

For whatever reason, only the agents nodes use ssd.  dbserver nodes insist on using the default storage (in this case, "standard")

I had to patch kubeconfig to switch the whole cluster to default on ssd in order for the dbserver to use ssd as persistent volume -  

This is the same behavior problem for arangodb in cluster mode deployed on GKE.

Desired state is for dbserver to  use storage defined in storageClass instead of just the default for the cluster.

**Comment by ajanikow** [comment]
Hello!

For ActiveFailover there only Agency and Single groups (https://www.arangodb.com/docs/stable/deployment-kubernetes-deployment-resource.html#specmode-string)

Best Regards,
Adam.

**Comment by mizunos** [comment]
Hi Adam,

I am aware of the configuration of Active Failover mode.  The issue reported is not about ActiveFailover.  It is about dbserver not taking Kubernetes defined  storage class ssd while the agents do.  This issue is seen in both Active Fail over mode and cluster mode for dbserver group.

I have to change the default storage class of the cluster to ssd in order dbserver to be provisioned using this storage definition while agents know to just use what is defined in storageClassName.

Hope this clarify this more

**Comment by ajanikow** [comment]
Hello!

I tested this for both scenarios and it works. 

For ActiveFailover you need to set storage class for agents and single groups (dbservers is just ignored). Just replaced dbservers to single and test passed. 

For Cluster you need to change it for agents and dbservers. I just used cluster mode with your example and it was fine. 

Both settings were applied properly - storage class and volume size.

Best Regards, 
Adam.

**Comment by mizunos** [comment]
Thx Adam. 

It is working good for new deployment.  

However I seemed to have trouble with updating an existing deployment via rolling upgrade per your suggestion on other email

What I did:
1 - update the deployment with dbservers.count: 4 and specified storageClassName: ssd (at the right indentation) for dbservers group
2 - re-apply the deployment (arango-cluster.yaml)

unfortunately nothing is happening.  The log seemed to say that it acknowledged the change but having trouble with executing it.

I sent all the operator logs to support.
